### PR TITLE
CFY-5760 fix missing property 'broker_ip'

### PR DESCRIPTION
**Pull request by adamlavie**

### CODE REVIEW

File: simple-manager-blueprint.yaml [hunk 1]
```diff
@@ -237,6 +237,8 @@ node_templates:
         target: manager_resources
       - type: cloudify.relationships.depends_on
         target: java_runtime
+      - type: cloudify.relationships.depends_on
```
Comment (dankilman) [inline simple-manager-blueprint.yaml]: add to others

File: components/manager/scripts/configure_manager.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ def _configure_security_properties():
 def main():
     if utils.is_upgrade:
         utils.create_upgrade_snapshot()
-    _disable_requiretty()
+    if not utils.is_upgrade and not utils.is_rollback:
```
Comment (dankilman) [inline components/manager/scripts/configure_manager.py]: if not is_rollback (add comment saying why)

File: components/elasticsearch/scripts/create.py [hunk 3]
```diff
@@ -405,22 +405,38 @@ def restore_upgrade_data(es_endpoint_ip, es_endpoint_port):
...
-    source = json.dumps(item['_source'])
-    metadata = _only_types(item, ['_type', '_id', '_index'])
+def _create_index_request(index):
+    source = json.dumps(index['_source'])
+    metadata = _only_types(index, ['_type', '_id', '_index'])
     action_and_meta_data = json.dumps({'index': metadata})
     return action_and_meta_data + os.linesep + source + os.linesep
 
 
+def _update_index_if_required(index):
```
Comment (dankilman) [inline components/elasticsearch/scripts/create.py]: element

File: components/elasticsearch/scripts/create.py [hunk 4]
```diff
@@ -405,22 +405,38 @@ def restore_upgrade_data(es_endpoint_ip, es_endpoint_port):
...
+    # Manager versions above 3.4.0 must contain a 'broker_ip' property in the
+    # agent configuration
+    if index['_type'] == 'provider_context':
+        content = index['_source']
+        agent_conf = content['context']['cloudify']['cloudify_agent']
```
Comment (dankilman) [inline components/elasticsearch/scripts/create.py]: read broker_ip runtime property from manager_configuration

File: components/elasticsearch/scripts/create.py [hunk 5]
```diff
@@ -405,22 +405,38 @@ def restore_upgrade_data(es_endpoint_ip, es_endpoint_port):
...
+        if not agent_conf.get('broker_ip') and utils.is_upgrade:
+            rabbit_props = utils.ctx_factory.get('rabbitmq')
+            broker_endpoint = rabbit_props.get('rabbitmq_endpoint_ip')
+            agent_conf['broker_ip'] = \
+                utils.get_rabbitmq_endpoint_ip(broker_endpoint)
+            index['_source']['context']['cloudify']['cloudify_agent'] = \
```
Comment (dankilman) [inline components/elasticsearch/scripts/create.py]: remove

File: components/manager/scripts/configure_manager.py [hunk 6]
```diff
@@ -74,7 +74,9 @@ def _configure_security_properties():
 def main():
     if utils.is_upgrade:
         utils.create_upgrade_snapshot()
-    _disable_requiretty()
+    # TTY has already been disabled in bootstrap
+    if not utils.is_upgrade and not utils.is_rollback:
```
Comment (adamlavie) [inline components/manager/scripts/configure_manager.py]: do it

**Pull request merged**